bosh-setup/azuredeploy.json

```diff
@@ -0,0 +1,341 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "location": {
+            "type": "string",
+            "allowedValues": [
+                "East Asia",
+                "Southeast Asia",
+                "Central US",
+                "East US",
+                "East US 2",
+                "West US",
+                "North Central US",
+                "South Central US",
+                "North Europe",
+                "West Europe",
+                "Japan West",
+                "Japan East",
+                "Brazil South",
+                "Australia East",
+                "Australia Southeast"
+            ],
+            "metadata": {
+                "description": "Location to deploy to"
+            }
+        },
+        "newStorageAccountName": {
+            "type": "string",
+            "metadata": {
+                "description": "Unique DNS Name for the Storage Account where the Virtual Machine's disks will be placed."
+            }
+        },
+        "virtualNetworkName": {
+            "type": "string",
+            "defaultValue": "boshvnet-crp",
+            "metadata": {
+                "description": "name of the virtual network" 
+            }
+        },
+        "subnetNameForBosh": {
+            "type": "string",
+            "defaultValue": "Bosh",
+            "metadata": {
+                "description": "name of the subnet for Bosh"
+            }
+        },
+        "subnetNameForCloudFoundry": {
+            "type": "string",
+            "defaultValue": "CloudFoundry",
```

Comment:
newStorageAccountName and vmName should be unique, so they are parameters.
Different vmSize can improve the performance.
virtualNetworkName, subnetNameForBosh, subnetNameForCloudFoundry and enableDNSOnDevbox are parameters for Bosh itself. Users can define them to customize their Bosh development environment.
In my opinion, they should be parameters instead of variables. And some of them have default values to reduce users' manual work.